Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -341,7 +339,13 @@ bool AsyncScrollingCoordinator::requestScrollToPosition(ScrollableArea& scrollab
     if (!stateNode)
         return false;
 
-    stateNode->setRequestedScrollData({ ScrollRequestType::PositionUpdate, scrollPosition, scrollType, clamping, animated });
+    if (options.animated == ScrollIsAnimated::Yes && options.scrollClass == ScrollClass::ScrollBy)
+        stateNode->setRequestedScrollData({ ScrollRequestType::DeltaUpdate, { }, scrollPosition - scrollableArea.scrollPosition(), options.type, options.clamping, options.animated });
```

Comment:
Not really a fan of computing the delta like this; it would be cleaner to just pass the delta down, but maybe that touches too many functions?

Reply:
Yeah this was the idea behind using the scroll class to indicate if the scroll to position was originally a delta or a position to scroll to. Rather than subtracting the current scroll position for the delta case, I think passing the original scroll delta down through `scrollToOptions` makes more sense.

Reply:
Added originalScrollDelta to `ScrollPositionChangeOptions`.